components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlinx/coroutines/flow/Flow.kt

```diff
@@ -47,6 +50,34 @@ fun <T, R> Flow<T>.ifChanged(transform: (T) -> R): Flow<T> {
     }
 }
 
+/**
+ * Returns a [Flow] containing only changed elements of the lists of the original [Flow].
+ *
+ * Example:
+ * ```
+ * Block: x -> x[0]  // Map to first character of input
+ * Original Flow: list("banana"), list("bandanna", "bus"), list("apple", "big", "coconut", "circle"), list("home")
+ * Mapped: ("b"), ("b", "b"), ("a", "b", "c", "c"), ("h")
+ * Returned Flow: "banana", "bus", "apple", "coconut", "circle", "home"
+ * ```
```

Comment:
Even though we came up with this method together: Just reading the example I have no clue what this does. :) It definitely took a while to understand what's going on.

Reply:
I wonder if we can come up with a better example that is at the same time concise.

Reply:
@pocmo OK, I'll spend time today to find a better example and adding more comments at least to explain why we should observe a change.

Reply:
@pocmo I've updated this now with more tests and two examples with more details. Can you take another look?